app/components/sidebars/main/channels_list/list/index.js

```diff
@@ -51,6 +53,8 @@ function mapStateToProps(state) {
         sidebarPrefs.favorite_at_top === 'true' && favoriteChannelIds.length,
     ));
 
+    const channelsByCategory = getChannelsByCategoryForCurrentTeam(state);
```

Comment:
Can potentially use config.EnableLegacySidebar here to prevent this action/reducer kicking in.

Reply:
absolutely use that to check if you need to execute the selector or not